source/PluginDev/Assets/GooglePlayGames/Platforms/Android/AndroidRealTimeMultiplayerClient.cs

```diff
@@ -34,111 +54,719 @@ public AndroidRealTimeMultiplayerClient(AndroidJavaObject account)
+                            {
+                                using (roomConfigBuilder.Call<AndroidJavaObject>("setVariant", (int) variant)) ;
+                            }
+
+                            using (var messageReceivedListener =
+                                new AndroidJavaObject(
+                                    "com.google.games.bridge.RealTimeMessageReceivedListenerProxy",
+                                    new MessageReceivedListenerProxy(listener)))
+                            using (roomConfigBuilder.Call<AndroidJavaObject>("setOnMessageReceivedListener",
+                                messageReceivedListener))
+                                ;
+
+                            using (var roomStatusUpdateCallback =
+                                new AndroidJavaObject("com.google.games.bridge.RoomStatusUpdateCallbackProxy",
+                                    new RoomStatusUpdateCallbackProxy(this, listener)))
+                            using (roomConfigBuilder.Call<AndroidJavaObject>("setRoomStatusUpdateCallback",
+                                roomStatusUpdateCallback))
+                                ;
+
+                            using (roomConfigBuilder.Call<AndroidJavaObject>("addPlayersToInvite",
+                                AndroidJavaConverter.ToJavaStringList(result.PlayerIdsToInvite)))
+                            {
+                                mRoomConfig = roomConfigBuilder.Call<AndroidJavaObject>("build");
+                            }
+
+                            mListener = listener;
+                        }
+
+                        using (var task = mClient.Call<AndroidJavaObject>("create", mRoomConfig))
+                        {
+                            AndroidTaskUtils.AddOnFailureListener(
+                                task,
+                                exception =>
+                                {
+                                    listener.OnRoomConnected(false);
+                                    CleanSession();
+                                });
+                        }
+                    }
+                });
+        }
+
+        private int GetMinParticipantsToStart()
+        {
+            int minParticipantsToStart;
+            using (var participants = mRoom.Call<AndroidJavaObject>("getParticipants"))
+            {
+                minParticipantsToStart = participants.Call<int>("size");
+            }
+
+            using (var autoMatchCriteria = mRoom.Call<AndroidJavaObject>("getAutoMatchCriteria"))
+            {
+                if (autoMatchCriteria != null)
+                {
+                    minParticipantsToStart = minParticipantsToStart +
+                                             autoMatchCriteria.Call<int>("getInt", "min_automatch_players", 0);
                 }
             }
 
-            using (var task = mClient.Call<AndroidJavaObject>("create", mRoomConfig))
+            if (mInvitation != null)
```

Comment:
> if (mInvitation != null)

why do we need this block?

Reply:
This method is totally removed. Instead, we have minPlayersToStart that is assigned when a room is created/joined.